apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.cs

```diff
@@ -62,6 +62,15 @@ internal SubscriberClientImpl(SubscriptionName subscriptionName, IEnumerable<Sub
         _modifyDeadlineSeconds = (int)((settings.AckDeadline ?? DefaultAckDeadline).TotalSeconds);
         var autoExtendInterval = TimeSpan.FromSeconds(_modifyDeadlineSeconds) - (settings.AckExtensionWindow ?? DefaultAckExtensionWindow);
         var autoExtendIntervalForExactlyOnceDelivery = settings.AckExtensionWindow ?? MinimumAckExtensionWindowForExactlyOnceDelivery;
+        // The value of 15 seconds ensures that we get ample time before the Ack deadline expires and that we don't send too frequent lease extensions.
+        var preferredExtendInterval = TimeSpan.FromSeconds(_modifyDeadlineSeconds - 15);
+        if (autoExtendIntervalForExactlyOnceDelivery > preferredExtendInterval)
```

Comment:
And this is comparing pears with apples:

autoExtendIntervelForExactlyOnceDelivery => window for extending the lease.
preferredExtendInterval => delay before we can attempt to extend the lease.

Reply:
And line 78 is also going to be wrong at least some of the time, i.e when autoExtendIntervelForExactlyOnceDelivery retains it's original value instead of being assigned preferredExtendInterval. In that case Math.Max value will be comparing a "windows for extending the lease" with a "delay before we can attempt to extend the lease", and those are complementary values not equivalents.

I wonder where else we have these same mismatches, and how these values are actually used.

Reply:
Thanks Amanda. This is now addressed.